publication-commons/src/main/java/no/unit/nva/publication/service/impl/ResourceMessages.java

```diff
@@ -1,27 +1,127 @@
+        Organization publisher = constructPublisher(mostRecentMessage);
+        EntityDescription entityDescription = constructEntityDescription(resourceTitleInMostRecentMessage);
+
+        return new Publication.Builder()
+                   .withOwner(mostRecentMessage.getOwner())
+                   .withPublisher(publisher)
+                   .withEntityDescription(entityDescription)
+                   .withIdentifier(mostRecentMessage.getResourceIdentifier())
+                   .build();
+    }
+
+    public List<MessageDto> getMessages() {
+        return messages;
+    }
+
+    public void setMessages(List<MessageDto> messages) {
         this.messages = messages;
     }
 
+    public Publication getPublication() {
+        return publication;
+    }
+
+    public void setPublication(Publication publication) {
+        this.publication = publication;
+    }
+
+
     @JacocoGenerated
-    public Resource getResource() {
-        return resource;
+    @Override
+    public int hashCode() {
+        return Objects.hash(getPublication(), getMessages());
     }
 
     @JacocoGenerated
-    public List<Message> getMessages() {
-        return messages;
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof ResourceMessages)) {
+            return false;
+        }
+        ResourceMessages that = (ResourceMessages) o;
+        return Objects.equals(getPublication(), that.getPublication()) && Objects.equals(getMessages(),
+            that.getMessages());
+    }
+
+    @JacocoGenerated
+    @Override
+    public String toString() {
+        return toJsonString();
+    }
+
+    private static ResourceMessages createNewResourceMessageInstance(List<Message> messages) {
+        messages.sort(ResourceMessages::oldestFirst);
+        Message mostRecentMessage = newestMessage(messages);
+        Publication publication = createPublicationDescription(mostRecentMessage);
+        return createResourceMessage(messages, publication);
+    }
+
+    private static Message newestMessage(List<Message> messages) {
+        return messages.get(messages.size() - 1);
+    }
+
+    private static int oldestFirst(Message o1, Message o2) {
+        return o1.getCreatedTime().compareTo(o2.getCreatedTime());
+    }
+
+    private static ResourceMessages createResourceMessage(List<Message> messages, Publication publication) {
+        ResourceMessages result = new ResourceMessages();
+        result.setPublication(publication);
+        result.setMessages(transformMessages(messages));
+        return result;
+    }
+
+    private static List<MessageDto> transformMessages(List<Message> messages) {
+        return messages.stream().map(MessageDto::fromMessage).collect(Collectors.toList());
+    }
+
+    private static EntityDescription constructEntityDescription(String resourceTitleInMostRecentMessage) {
```

Comment:
Argument name, is it important to annotate what the title is, or just that it is a title? I think the latter at this point in the code.